src/go/rpk/pkg/cli/transform/logs.go

```diff
@@ -0,0 +1,470 @@
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.uber.org/zap"
+)
+
+var (
+	minTime = time.Unix(0, 0).UTC()
+	maxTime = time.Unix(math.MaxInt64, math.MaxInt64).UTC()
+)
+
+func newLogsCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var follow bool
+	var head, tail int
+	var since, until timequery
+	since.time = minTime
+	until.time = maxTime
+	cmd := &cobra.Command{
+		Use:     "logs [NAME]",
+		Aliases: []string{"log"},
+		Short:   "View logs for a transform",
+		Args:    cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			f := p.Formatter
+			if h, ok := f.Help([]string{}); ok {
+				out.Exit(h)
+			}
+
+			transformName := args[0]
+
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			admin, err := kafka.NewAdmin(fs, p)
+			out.MaybeDie(err, "unable to create admin client: %v", err)
+
+			topics, err := admin.ListTopics(cmd.Context(), "_redpanda.transform_logs")
+			out.MaybeDie(err, "unable to get logs topic: %v", err)
+			topic := topics.TopicsList()[0]
+			partition := computeLogPartition(transformName, topic)
+
+			startOffset := int64(-2) // NewOffset.At treats this as start
+			if !since.time.Equal(minTime) {
+				offsets, err := admin.ListOffsetsAfterMilli(
+					cmd.Context(),
+					since.time.UnixMilli(),
+					topic.Topic,
+				)
+				out.MaybeDie(err, "unable to list offsets: %v", err)
+				offset, ok := offsets.Lookup(topic.Topic, partition)
+				if !ok {
+					out.Die("unable to list offset for partition: %d", partition)
+				}
+				startOffset = offset.Offset
+			}
+			maxOffset := int64(math.MaxInt64)
+			if !follow {
+				offsets, err := admin.ListCommittedOffsets(cmd.Context(), topic.Topic)
+				out.MaybeDie(err, "unable to list offsets: %v", err)
+				offset, ok := offsets.Lookup(topic.Topic, partition)
+				if !ok {
+					out.Die("unable to list offset for partition: %d", partition)
+				}
+				// The max offset here is exclusive of the last record, so minus one so that we stop once we hit that last record.
+				maxOffset = offset.Offset - 1
+				// If we are starting after our max offset we need to short circuit
+				if !until.time.Equal(maxTime) {
+					untilOffsets, err := admin.ListOffsetsAfterMilli(
+						cmd.Context(),
+						until.time.UnixMilli(),
+						topic.Topic,
+					)
+					out.MaybeDie(err, "unable to list offsets: %v", err)
+					untilOffset, ok := untilOffsets.Lookup(topic.Topic, partition)
+					if !ok {
+						out.Die("unable to list offset for partition: %d", partition)
+					}
+					maxOffset = min(maxOffset, untilOffset.Offset)
+				}
+			}
+			zap.L().Sugar().Infof("reading logs topic with bounds [%v, %v] on partition %d", startOffset, maxOffset, partition)
```

Comment:
[question] Why INFO here instead of debug? I _think_ these will appear in stoud even w/o the `-v` flags, the difference between these and fmt.Println() is the formatting.

Maybe it will be weird if you choose `--format json` and see a Log + JSON string.

Reply:
These were mostly for debugging. I'll move to debug